lib/node_modules/@stdlib/_tools/remark/plugins/remark-lint-html-section-structure/lib/main.js

```diff
@@ -0,0 +1,367 @@
+		var match;
+		var data;
+
+		// Store position for tracking purposes:
+		if ( node.position ) {
+			positions[ node.position.start.line ] = node;
+		}
+
+		// Check if this is a combined section end with comment on the same line (which is an error):
+		combinedMatch = RE_SECTION_END_WITH_COMMENT.exec( node.value );
+		if ( combinedMatch ) {
+			debug( 'Found combined section end with comment: %s', node.value );
+			msg = 'Missing empty line between closing section tag and comment. There should be an empty line between </section> and <!-- /.' + combinedMatch[1] + ' -->';
+			errCode = 'missing-empty-line';
+			reportErr( file, node, msg, errCode );
+
+			// Make sure to pop the section stack for this case - otherwise we'll get "unclosed section" errors:
+			if ( sectionStack.length > 0 ) {
+				debug( 'Popping section stack for combined section end+comment' );
+				sectionStack.pop();
+			}
+			return;
+		}
+
+		// Check if this is a section start tag:
+		sectionMatch = RE_SECTION_START.exec( node.value );
+		if ( sectionMatch ) {
+			debug( 'Found a section start tag with class: %s', sectionMatch[1] || '[none]' );
+
+			data = {
+				'node': node,
+				'className': sectionMatch[1] || '',
+				'position': node.position,
+				'children': [],
+				'index': index,
+				'parent': parent
+			};
+
+			if ( sectionStack.length === 0 ) {
+				// This is a root level section...
+				rootSections.push( data );
+				structure.push( data );
+			} else {
+				// This is a nested section, add to parent's children:
+				sectionStack[ sectionStack.length - 1 ].children.push( data );
+			}
+
+			sectionStack.push( data );
+			return;
+		}
+
+		// Check if this is a section end tag:
+		endMatch = RE_SECTION_END.exec( node.value );
+		if ( endMatch ) {
+			debug( 'Found a section end tag' );
+
+			if ( sectionStack.length === 0 ) {
+				debug( 'Orphaned section end tag found' );
+				msg = 'Orphaned section closing tag. Found </section> without a matching opening <section> tag. Remove this tag or add a corresponding opening tag.';
+				errCode = 'orphaned-section-end';
+				reportErr( file, node, msg, errCode );
+				return;
+			}
+
+			// Special handling for direct closing tag followed by a comment (no text node between)...
+			if ( index + 1 < parent.children.length ) {
+				nextNode = parent.children[ index + 1 ];
+
+				// Look at the next node to check if it's a comment:
+				if ( nextNode.type === 'html' ) {
+					// Check if the next node is a comment but NOT on the same line
+					commentClass = extractCommentClass( nextNode, RE_END_COMMENT, RE_END_COMMENT_SIMPLE );
+
+					if ( commentClass ) {
+						// Check if there are at least 2 line breaks between them:
+						if ( !hasValidWhitespace( node, nextNode ) ) {
+							msg = 'Missing proper empty line after closing section tag. There should be an empty line between </section> and the closing comment.';
+							errCode = 'missing-empty-line-after-section';
+							reportErr( file, node, msg, errCode );
```

Comment:
@Planeshifter Just checking, but for a number of these `reportErr` statements, we don't early return, but instead fall through to subsequent logic. Are these fall-throughs intentional? I'd want to avoid a scenario where lint errors pile up and it becomes confusing for devs to understand what the actual issue is which is triggering lint failures.

Reply:
If we shouldn't be early returning, it may be good to add a comment in each circumstance indicating that fall-through is intentional.

Reply:
@kgryte Double-checked, left some comments, and added an early return to make the code more readable.